### PR TITLE
Basic auth

### PR DESCRIPTION
**Pull request by pierrotsmnrd**
Basic auth now works, thanks to succeeding setting a cookie.

### CODE REVIEW

File: ragna/_ui/logout_page.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        self.api_wrapper.auth_token = None
+
+    def __panel__(self):
+        # Usually, we do a redirect this way :
+        # >>> pn.state.location.param.update(reload=True, pathname="/")
+        # But it only works once the page is fully loaded.
+        # So we render a javascript redirect instead.
+        return pn.pane.HTML(
+            """<script>
+                            document.cookie = "auth_token=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/";
```
Comment (pmeier) [inline ragna/_ui/logout_page.py:19]: What is the expiration in the past here?

File: ragna/_ui/auth_page.py [hunk 2]
```diff
@@ -49,10 +51,17 @@ async def perform_login(self, event=None):
...
                 self.login_input.value, self.password_input.value
             )
-        except Exception:
+
+            if authed:
+                # Sets the cookie on the JS side
+                self.custom_js = f""" document.cookie = "auth_token={self.api_wrapper.auth_token}; path:/";  """
+
+        except Exception as e:
+            print(e)
```
Comment (pmeier) [inline ragna/_ui/auth_page.py:60]: Remove?

```suggestion
```

**Review by pmeier — APPROVED** [review]
Two minor comments, but otherwise nothing stood out. I didn't have time to properly test this, but since it is only demo auth, I'm ok with going the quick way for now.

In the future, we need to make the UI auth code part of the demo auth class. It should be together since the code on the UI, i.e. the username / password form, very much relies on the auth that is implemented on the API.

**Comment by pmeier** [comment]
@pierrotsmnrd I dismissed my review, because I couldn't get the auth to work locally. I had it this morning, but now after starting the API and trying to visit the UI, I see this

<details><summary>Traceback</summary>
<p>

```
Uncaught exception GET / (127.0.0.1)
HTTPServerRequest(protocol='http', host='localhost:31477', method='GET', uri='/', version='HTTP/1.1', remote_ip='127.0.0.1')
Traceback (most recent call last):
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/tornado/web.py", line 1786, in _execute
    result = await result
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/panel/io/server.py", line 519, in get
    session = await self.get_session()
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/panel/io/server.py", line 474, in get_session
    session = await super().get_session()
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/bokeh/server/views/session_handler.py", line 145, in get_session
    session = await self.application_context.create_session_if_needed(session_id, self.request, token)
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/bokeh/server/contexts.py", line 240, in create_session_if_needed
    self._application.initialize_document(doc)
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/panel/io/server.py", line 410, in initialize_document
    super().initialize_document(doc)
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/bokeh/application/application.py", line 190, in initialize_document
    h.modify_document(doc)
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/bokeh/application/handlers/function.py", line 140, in modify_document
    self._func(doc)
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/panel/io/server.py", line 149, in _eval_panel
    panel = panel()
  File "/home/philip/git/ora/ragna/_ui/app.py", line 74, in index_page
    main_page = MainPage(api_wrapper=self.api_wrapper, template=template)
  File "/home/philip/git/ora/ragna/_ui/main_page.py", line 39, in __init__
    self.refresh_data()
  File "/home/philip/git/ora/ragna/_ui/main_page.py", line 42, in refresh_data
    self.chats = self.api_wrapper.get_chats()
  File "/home/philip/git/ora/ragna/_ui/api_wrapper.py", line 43, in get_chats
    json_data = self.client.get("/chats").raise_for_status().json()
  File "/home/philip/.conda/envs/ragna-dev/lib/python3.9/site-packages/httpx/_models.py", line 749, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Client error '401 Unauthorized' for url 'http://127.0.0.1:31476/chats'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
500 GET / (127.0.0.1) 41.66ms
```


</p>
</details>

It seems that the auth page is not hit first and thus any other request, e.g. getting the chats for a given user, fails.

File: ragna/_ui/logout_page.py (hunk 1 above)
Comment (pierrotsmnrd) [reply ragna/_ui/logout_page.py:21]: I was hoping the context would be explicit enough but I've pushed a comment to clarify : to remove a cookie, you have to set its expiry date to the past to force it to expire.

**Comment by pierrotsmnrd** [comment]
@pmeier the problem is that the auth token is invalid after restarting the API. 
I've covered that case actually, but ... in my other PR (#135), as it was part of a refactor of ApiWrapper.

I'm closing this one, PR #137 supersedes.

**Pull request closed without merging**